platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/snapshot/SignedStateFileReader.java

```diff
@@ -87,43 +104,104 @@ public static List<SavedStateInfo> getSavedStateFiles(
+                    (final MerkleDataInputStream in) -> {
+                        final int fileVersion = readAndCheckSigSetFileVersion(in);
                         final SigSet sigSet = in.readSerializable();
-                        return new StateFileData(state, hash, sigSet);
-                    } catch (final IOException e) {
-                        throw new IOException("Failed to read snapshot file " + stateFile.toFile(), e);
-                    }
-                });
+                        return new StateFileData(data.state, data.hash, sigSet, fileVersion);
+                    });
+        } else {
+            normalizedData = data;
+        }
 
         final SignedState newSignedState = new SignedState(
                 platformContext,
                 CryptoStatic::verifySignature,
-                data.state(),
+                normalizedData.state,
                 "SignedStateFileReader.readStateFile()",
                 false,
                 false,
                 false);
 
-        newSignedState.setSigSet(data.sigSet());
+        newSignedState.setSigSet(normalizedData.sigSet);
 
         returnState = new DeserializedSignedState(
-                newSignedState.reserve("SignedStateFileReader.readStateFile()"), data.hash());
+                newSignedState.reserve("SignedStateFileReader.readStateFile()"), normalizedData.hash);
 
         return returnState;
     }
 
+    /**
+     * Reads a SignedState from disk using the provided snapshot reader function.
+     * @param stateFile the file to read from
+     * @param snapshotStateReader state snapshot reading function
+     * @return a signed state with it's associated hash (as computed when the state was serialized)
+     * @throws IOException if there is any problems with reading from a file
+     */
+    @NonNull
+    public static StateFileData readStateFileData(
+            @NonNull final Path stateFile,
+            @NonNull final CheckedBiFunction<MerkleDataInputStream, Path, MerkleRoot, IOException> snapshotStateReader)
+            throws IOException {
+        return deserializeAndDebugOnFailure(
+                () -> new BufferedInputStream(new FileInputStream(stateFile.toFile())),
+                (final MerkleDataInputStream in) -> {
+                    final int fileVersion = readAndCheckStateFileVersion(in);
+
+                    final Path directory = stateFile.getParent();
+                    if (fileVersion == INIT_STATE_FILE_VERSION) {
+                        return readStateFileDataV1(stateFile, snapshotStateReader, in, directory, fileVersion);
+                    } else if (fileVersion == SIG_SET_SEPARATE_STATE_FILE_VERSION) {
+                        return createStateFileDataV2(stateFile, snapshotStateReader, in, directory, fileVersion);
+                    } else {
+                        throw new IOException("Unsupported protocol version: " + fileVersion);
```

Comment:
wrong exception message, should be `Unsupported state file version`

Reply:
Fixed